pkg/logql/syntax/lex.go

```diff
@@ -138,7 +138,7 @@ type lexer struct {
 	builder strings.Builder
 }
 
-func (l *lexer) Lex(lval *exprSymType) int {
+func (l *lexer) Lex(lval *syntaxSymType) int {
```

Comment:
Prefix changes from `expr` to `sytnax`, because the prefix is declared by the name of the `.y` file, see `yacc` target in Makefile.